src/context-selection/org-unit-selector-bar-item/use-load-offline-levels.js

```diff
@@ -0,0 +1,43 @@
+import { useDataEngine } from '@dhis2/app-runtime'
+import { useEffect, useState } from 'react'
+import loadOfflineLevel from './load-offline-level.js'
+import useOfflineLevelsToLoad from './use-offline-levels-to-load.js'
+import useOrganisationUnitLevels from './use-organisation-unit-levels.js'
+
+/**
+ * As the service worker caches request responses,
+ * it should suffice to simply perform all request the org unit tree would
+ * perform as well in advance
+ */
+export default function useLoadOfflineLevels() {
+    const [loading, setLoading] = useState(true)
+    const [error, setError] = useState(undefined)
+    const [data, setData] = useState(false)
+
+    const dataEngine = useDataEngine()
+    const organisationUnitLevels = useOrganisationUnitLevels()
+    const offlineLevelsToLoad = useOfflineLevelsToLoad(organisationUnitLevels)
+    const { data: offlineLevelsToLoadData } = offlineLevelsToLoad
+
+    useEffect(() => {
+        // Can't pass async function to useEffect
+        offlineLevelsToLoadData &&
+            (async () => {
+                await Promise.all(
+                    offlineLevelsToLoadData.map((offlineLevelToLoad) => {
+                        const payload = {
+                            dataEngine,
+                            ...offlineLevelToLoad,
+                        }
+
+                        return loadOfflineLevel(payload)
+                    })
+                )
+                    .then(() => setData(true))
+                    .catch(setError)
+                    .finally(() => setLoading(false))
+            })()
+    }, [dataEngine, offlineLevelsToLoadData, setLoading, setData])
+
+    return { loading, error, data }
```

Comment:
Since `data` is always just a boolean representing if offline levels have been successfully loaded, maybe this makes more sense:

```suggestion
    return { fetching, error, offlineLevelsLoaded }
```

Reply:
the hook just returns `true` now. I actually wanted it to return nothing as the return value wasn't being used, but it's required for the jest test to work with `@testing-library/react-hooks`'s `withForNextUpdate`